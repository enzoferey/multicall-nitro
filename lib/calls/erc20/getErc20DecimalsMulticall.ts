import type { Call, CallOptions } from "../../types";

export function getErc20DecimalsMulticall(
  erc20ContractAddress: string,
  options: CallOptions
): Call {
  const { label } = options;

  return {
    target: erc20ContractAddress,
    call: ["decimals()(uint8)"],
    label,
  };
}
