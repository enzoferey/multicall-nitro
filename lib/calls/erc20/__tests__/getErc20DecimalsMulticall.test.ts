import { describe, it, expect } from "vitest";

import { getErc20DecimalsMulticall } from "../getErc20DecimalsMulticall";

describe("getErc20DecimalsMulticall", () => {
  it("should match the call", () => {
    expect(
      getErc20DecimalsMulticall("0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee", {
        label: "some label",
      })
    ).toMatchSnapshot();
  });
});
