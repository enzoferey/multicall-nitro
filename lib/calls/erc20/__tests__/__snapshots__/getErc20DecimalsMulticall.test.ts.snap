// Vitest Snapshot v1

exports[`getErc20DecimalsMulticall > should match the call 1`] = `
{
  "call": [
    "decimals()(uint8)",
  ],
  "label": "some label",
  "target": "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee",
}
`;
